### PR TITLE
1 step : Creation of teams

### PR DESCRIPTION
**Pull request by DrKrakus**

### CODE REVIEW

File: oc-game-swift.xcodeproj/xcuserdata/noox.xcuserdatad/xcschemes/xcschememanagement.plist [hunk 1]
```diff
@@ -9,6 +9,11 @@
 			<key>orderHint</key>
 			<integer>0</integer>
 		</dict>
+		<key>oc-game-swift.xcscheme</key>
+		<dict>
+			<key>orderHint</key>
```
Comment (jhanzo) [inline oc-game-swift.xcodeproj/xcuserdata/noox.xcuserdatad/xcschemes/xcschememanagement.plist]: why do you add this variable in this `plist` ?

File: oc-game-swift/Game/Game.swift [hunk 2]
```diff
@@ -7,3 +7,106 @@
 //
 
 import Foundation
+
+// Create the Game class
+class Game {
+    static var playerList = [Player]()
```
Comment (jhanzo) [inline oc-game-swift/Game/Game.swift]: it is better to declare type and define a default value : 
```swift
static var playerList: [Player] = []

File: oc-game-swift/Game/Game.swift [hunk 3]
```diff
@@ -7,3 +7,106 @@
...
+    
+    // Show credits
+    static func showCredits() {
+        print("This awesome game is developped by Jérôme Krakus !")
+    }
+    
+    // Create players
+    static func createPlayers() {
+        // While there is no 2 players for the game
+        while Game.playerList.count < 2 {
```
Comment (jhanzo) [inline oc-game-swift/Game/Game.swift]: if you plan to add more users next, maybe you could think about another solution ?

File: oc-game-swift/Game/Game.swift [hunk 4]
```diff
@@ -7,3 +7,106 @@
...
+            print("---------------------------------------------------"
+                + "\nChoose your name player \(Game.playerList.count + 1)"
+                + "\n---------------------------------------------------")
+            // Read the name of player
+            if let name = readLine(){
+                if name != "" {
```
Comment (jhanzo) [inline oc-game-swift/Game/Game.swift]: you could use `isEmpty` property : 
```swift
if !name.isEmpty {

File: oc-game-swift/Game/Game.swift [hunk 5]
```diff
@@ -7,3 +7,106 @@
...
+                    // If the name is not empty, create and add player to the game
+                    Game.playerList.append(Player(playerName: name))
```
Comment (jhanzo) [inline oc-game-swift/Game/Game.swift]: `playerName` property is redundant, you could rename it with `name`

File: oc-game-swift/Game/Game.swift [hunk 6]
```diff
@@ -7,3 +7,106 @@
...
+                } else {
+                    // If not
```
Comment (jhanzo) [inline oc-game-swift/Game/Game.swift]: if there is not other comments you could remove this line

File: oc-game-swift/Game/Game.swift [hunk 7]
```diff
@@ -7,3 +7,106 @@
...
+                    print("Dont be like that... Choose a name 🙃")
+                }
+            }
+        }
+        // When 2 players are created
+        pickAHeroMenu()
+        
+    }
+    
+    // Mark : - Hero selection
```
Comment (jhanzo) [inline oc-game-swift/Game/Game.swift]: Mark should be written like this : 
```swift
// MARK:

File: oc-game-swift/Game/Game.swift [hunk 8]
```diff
@@ -7,3 +7,106 @@
...
+    static func pickAHeroMenu() {
+        // Each player choose 3 heroes
+        for player in Game.playerList {
+            // While one player doesnt have 3 hero
+            while player.heroTeam.count < 3 {
```
Comment (jhanzo) [inline oc-game-swift/Game/Game.swift]: the number '3' could be put in a configuration file, eg. Config.swift

File: oc-game-swift/Game/Game.swift [hunk 9]
```diff
@@ -7,3 +7,106 @@
...
+                print("\(player.playerName) choose \(3 - player.heroTeam.count) heroes"
+                    + "\n 1. Fighter - Basic, strong, your best friend here"
```
Comment (jhanzo) [inline oc-game-swift/Game/Game.swift]: Maybe you could put description in each Hero class for avoiding to write them here

File: oc-game-swift/Game/Game.swift [hunk 10]
```diff
@@ -7,3 +7,106 @@
...
+                    + "\n 2. Healer - You should really think about him for your team"
+                    + "\n 3. Dwarf - Fearless and powerfull, but weak too"
+                    + "\n 4. Colossus - Tanky as fuck, but make less damage than your little sister")
+                
+                // Check the awnser
+                if let choice = readLine() {
+                    // If the name is correct
+                    if choice != "" {
```
Comment (jhanzo) [inline oc-game-swift/Game/Game.swift]: `.isEmpty`

File: oc-game-swift/Heroes/Healer.swift [hunk 11]
```diff
@@ -8,4 +8,6 @@
 
 import Foundation
 
-
+// Healer class
+class Healer: Hero {
```
Comment (jhanzo) [inline oc-game-swift/Heroes/Healer.swift]: you could add property in these classes like `description`

File: oc-game-swift/Heroes/HeroType.swift [hunk 12]
```diff
@@ -8,3 +8,10 @@
 
 import Foundation
 
+// List the different type of heroes
+enum HeroType {
```
Comment (jhanzo) [inline oc-game-swift/Heroes/HeroType.swift]: Nice 👍

File: oc-game-swift/Players/Player.swift [hunk 13]
```diff
@@ -8,3 +8,55 @@
 
 import Foundation
 
+// Create the Player class
+class Player {
+    // Player name
+    var playerName: String
+    // Player's team
+    var heroTeam = [Hero]()
```
Comment (jhanzo) [inline oc-game-swift/Players/Player.swift]:
```swift 
var heroTeam: [Hero] = []

File: oc-game-swift/Players/Player.swift [hunk 14]
```diff
@@ -8,3 +8,55 @@
...
+    
+    // Init with name
+    init(playerName: String) {
```
Comment (jhanzo) [inline oc-game-swift/Players/Player.swift]: Simplify parameter name : 
```swift
init(name: String)

**Review by jhanzo — CHANGES_REQUESTED** [review]

File: oc-game-swift.xcodeproj/xcuserdata/noox.xcuserdatad/xcschemes/xcschememanagement.plist (hunk 1 above)
Comment (DrKrakus) [reply oc-game-swift.xcodeproj/xcuserdata/noox.xcuserdatad/xcschemes/xcschememanagement.plist]: I think xcode added that when i've changed the name of the project.

File: oc-game-swift/Game/Game.swift (hunk 2 above)
Comment (DrKrakus) [reply oc-game-swift/Game/Game.swift]: Done

File: oc-game-swift/Game/Game.swift (hunk 4 above)
Comment (DrKrakus) [reply oc-game-swift/Game/Game.swift]: Done

File: oc-game-swift/Game/Game.swift (hunk 10 above)
Comment (DrKrakus) [reply oc-game-swift/Game/Game.swift]: Done

File: oc-game-swift/Players/Player.swift (hunk 13 above)
Comment (DrKrakus) [reply oc-game-swift/Players/Player.swift]: Done

File: oc-game-swift/Players/Player.swift (hunk 14 above)
Comment (DrKrakus) [reply oc-game-swift/Players/Player.swift]: Done

File: oc-game-swift/Heroes/Healer.swift (hunk 11 above)
Comment (DrKrakus) [reply oc-game-swift/Heroes/Healer.swift]: I will complete hero classes with the phase 2 : the fight.

**Pull request merged**